xarray/core/dataset.py

```diff
@@ -941,25 +941,6 @@ def _replace_with_new_dims(
             variables, coord_names, dims, attrs, indexes, inplace=inplace
         )
 
-    def _replace_vars_and_dims(
-        self,
-        variables: Dict[Hashable, Variable],
-        coord_names: set = None,
-        dims: Dict[Hashable, int] = None,
-        attrs: Union[Dict[Hashable, Any], None, Default] = _default,
-        inplace: bool = False,
-    ) -> "Dataset":
-        """Deprecated version of _replace_with_new_dims().
-
-        Unlike _replace_with_new_dims(), this method always recalculates
-        indexes from variables.
-        """
-        if dims is None:
-            dims = calculate_dimensions(variables)
-        return self._replace(
-            variables, coord_names, dims, attrs, indexes=None, inplace=inplace
```

Comment:
looks like this old function set `indexes=None` so your later changes make sense.

Also see docstring: 
`Unlike _replace_with_new_dims(), this method always recalculates indexes from variables.`